File: docs/competitors/JavaScript/CircleSquare/CircleSquare.js
const readline = require('readline'); // импорт модуля nodeJS, который позволяет читать строки с клавиатуры

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Введите радиус круга: ', (input) => {
    const r = parseFloat(input);

    if (isNaN(r)) {
        console.log('Ошибка: введено не число.');
    } else if (r < 0) {
        console.log('Ошибка: радиус не может быть отрицательным.');
    } else {
        const area = Math.PI * r * r;
        console.log(`Площадь круга = ${area.toFixed(6)}`);
    }

    rl.close(); // закрывает интерфейс
});

File: docs/competitors/python/CircleSquare/CircleSquare.py
import math

try:
    r_input = input("Введите радиус круга: ")
    r = float(r_input)
except ValueError:
    print("Ошибка: введено не число.")
else:
    if r < 0:
        print("Ошибка: радиус не может быть отрицательным.")
    else:
        area = math.pi * r ** 2
        print(f"Площадь круга = {area:.6f}")